pkg/sidecar/sidecar.go

```diff
@@ -460,7 +469,7 @@ func (s *Sidecar) CheckLiveness() bool {
 			return false
 		}
 	}
-	return s.health.FileWriteStatuses.X509WriteStatus != writeStatusFailed
+	return s.health.FileWriteStatuses.X509WriteStatus != nil && *s.health.FileWriteStatuses.X509WriteStatus != writeStatusFailed
```

Comment:
If this is nil it will fail. Should it match what `CheckReadiness()` does?

Reply:
I believe I was concerned about if this function got called before the health check stuff was isetup (`setupHealth()`). Without the nil check, `*s.health.FileWriteStatuses.X509WriteStatus != writeStatusFailed` would be true in such a case.

Reply:
If only jwt is configured but not x509, the liveness check would never pass.

Reply:
```suggestion
        if s.x509Enabled() {
            if s.health.FileWriteStatuses.X509WriteStatus == nil || *s.health.FileWriteStatuses.X509WriteStatus == writeStatusFailed {
                return false
            }
        }
	return true
```

Reply:
I think this would fix it ^^

Reply:
> If only jwt is configured but not x509, the liveness check would never pass.

Ah, I was thinking `X509WriteStatus` would be `nil`, but `updateCertificates()` is always called it appears even if no X509s are configured. Maybe `updateCertificates()` should have an `if s.x509Enabled()` around it?

Reply:
We gate the call to `updateCertificates()` in `RunDaemon()` by not setting up the watcher